### PR TITLE
SMB Session

### PR DESCRIPTION
**Pull request by mbscit**

### CODE REVIEW

File: smb/pom.xml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  ~ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  ~ GNU General Public License for more details.
+  -->
+
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>ch.cyberduck</groupId>
+        <artifactId>parent</artifactId>
+        <version>8.5.7-SNAPSHOT</version>
```
Comment (dkocher) [inline smb/pom.xml:21]: Fix parent version.

**Review by dkocher — CHANGES_REQUESTED** [review]
- Add implementation of `AbstractExceptionMappingService<SMBApiException>` mapping errors to various subclasses of `BackgroundException`.

**Review by ylangisc — COMMENTED** [review]
* consistent use of `final` for method parameters and local variables
* would be great to concentrate on writing tests as a next step

**Comment by AliveDevil** [comment]
This is superseded by #14672?

**Pull request closed without merging**